src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObj: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemKey = `${item[key]}`;
+
+    if (groupedObj[itemKey]) {
+      groupedObj[itemKey].push(item);
+    } else {
+      groupedObj[itemKey] = [item];
+    }
```

Comment:
you can invert your condition to just add blank array, and then push your item to get rid of `else`

Reply:
ok